File: 6.adapter/adapter.ts
// Interfaz objetivo
interface PaymentProcessor {
    processPayment(details: string): void;
}

// Clase existente incompatible
class LegacyPaymentSystem {
    public makePayment(data: string): void {
        console.log(`Pago procesado: ${data}`);
    }
}

// Adaptador
class PaymentAdapter implements PaymentProcessor {
    private legacySystem: LegacyPaymentSystem;

    constructor(legacySystem: LegacyPaymentSystem) {
        this.legacySystem = legacySystem;
    }

    public processPayment(details: string): void {
        this.legacySystem.makePayment(details);
    }
}

// Uso
const legacySystem = new LegacyPaymentSystem();
const adapter: PaymentProcessor = new PaymentAdapter(legacySystem);
adapter.processPayment("Detalles del pago");

File: 6.adapter/adapter.py
# Clase existente incompatible
class LegacyPaymentSystem:
    def make_payment(self, data):
        print(f"Pago procesado: {data}")

# Interfaz objetivo
class PaymentProcessor:
    def process_payment(self, details):
        pass

# Adaptador
class PaymentAdapter(PaymentProcessor):
    def __init__(self, legacy_system):
        self.legacy_system = legacy_system

    def process_payment(self, details):
        self.legacy_system.make_payment(details)

# Uso
legacy_system = LegacyPaymentSystem()
adapter = PaymentAdapter(legacy_system)

adapter.process_payment("Detalles del pago")
